client/components/AllProducts.js

```diff
@@ -2,32 +2,60 @@ import React from 'react';
 import { connect } from 'react-redux';
 import { fetchProducts } from '../store/products';
 import { Link } from 'react-router-dom';
+import FilterInput from './FilterInput';
 
 class AllProducts extends React.Component {
-    constructor(props) {
-        super(props)
-    }
-    componentDidMount() {
-        this.props.fetchInitialData();
-    }
-    render() {
-        return (
-            <div>
-                <ul>
-                    {this.props.products && this.props.products.map(product =>
-                        (<li key={product.id}>
-                            <Link to={`/products/${product.id}`}>{product.title}</Link>
-                        </li>)
-                    )}
-                </ul>
-            </div>
-        )
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      inputValue: '',
     }
+
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(event) {
+    const value = event.target.value;
+    this.setState({
+      inputValue: value,
+    });
+  }
+
+  componentDidMount() {
+    this.props.fetchInitialData();
+  }
+
+  render() {
+    const { products } = this.props;
+    const inputValue = this.state.inputValue;
+    const re = new RegExp(inputValue, 'i');
```

Comment:
we could be a bit more explicit here and call the variable `regex`  instead of `re` ;-)